python/src/trezorlib/ethereum.py

```diff
@@ -141,24 +157,181 @@ def encode_data(value: Any, type_name: str) -> bytes:
     raise ValueError(f"Unsupported data type for direct field encoding: {type_name}")
 
 
+def download_from_url(url: str) -> bytes:
+    try:
+        r = requests.get(url)
+        r.raise_for_status()
+        return r.content
+    except requests.exceptions.HTTPError as err:
+        raise RuntimeError(err.response)
+
+
+def get_network_definition_url(
+    chain_id: Optional[int] = None, slip44: Optional[int] = None
+) -> str:
+    if not ((chain_id is None) != (slip44 is None)):  # not XOR
+        raise ValueError(
+            "Exactly one of chain_id or slip44 parameters are needed to construct network definition url."
+        )
+
+    if chain_id is not None:
+        return DEFS_URL_LOOKUP_TEMPLATE.format(
+            lookup_type=DEFS_NETWORK_BY_CHAINID_LOOKUP_TYPE,
+            id=chain_id,
+            name=DEFS_NETWORK_URI_NAME,
+        )
+    else:
+        return DEFS_URL_LOOKUP_TEMPLATE.format(
+            lookup_type=DEFS_NETWORK_BY_SLIP44_LOOKUP_TYPE,
+            id=slip44,
+            name=DEFS_NETWORK_URI_NAME,
+        )
+
+
+def get_token_definition_url(chain_id: int = None, token_address: str = None) -> str:
```

Comment:
Types - `Optional[int]`, `Optional[str]`

However, seeing the `ValueError` below enforcing none of these is `None`, we could strictly enforce `int` and `str` types and get rid of the `ValueError`. (the only caller is itself enforcing `chain_id is not None and token_address is not None`)

Reply:
code was removed